### PR TITLE
DEPLOY-240 : ADR for Using One DB instance per Component

### PR DESCRIPTION
**Pull request by svidrascu**
ADR added for Decision to use one DB for current and new components that need to store data in databases

### CODE REVIEW

File: docs/adrs/0008-using-one-db-per-service.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
+# 7. Using one database per service
```
Comment (gavincornwell) [inline docs/adrs/0008-using-one-db-per-service.md]: This should be 8.

File: docs/adrs/0008-using-one-db-per-service.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+Date: 2018-01-10
+
+## Status
+
+Proposed
+
+## Context
+
+Many of the components the digital business platform offers need a database for storing information.
+Currently Content Services, Process Services and Sync Service do this but there might be more to come in the future. All three are using the same version of the database so the obvious question arrised: Can we just use one database for all our services?
```
Comment (gavincornwell) [inline docs/adrs/0008-using-one-db-per-service.md]: "was raised" instead of "arrised"

File: docs/adrs/0008-using-one-db-per-service.md [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+We identified advantages of using one database per service or one for all the services:
```
Comment (gavincornwell) [inline docs/adrs/0008-using-one-db-per-service.md]: In the last ADR I did Ray asked me to change bullets/lists to a sentence as it's easier to read (and is a decision the architects group made for consistency and readability).

File: docs/adrs/0008-using-one-db-per-service.md [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+Best practise in microservices world (decoupling)
+Allows any service to move to a different version of PostgreSQL or change persistence layer
+Stops one service from being able to slow down the DB, impacting performance on other services
+
+One database for all services:
+Less components to configure, keep track of, monitor and backup
+
+## Decision
+
+We will proceed with the one database per service for existent and future components as there are a number of advantages over having all our services fight over the same database.
```
Comment (gavincornwell) [inline docs/adrs/0008-using-one-db-per-service.md]: "existing" instead of "existent"

File: docs/adrs/0008-using-one-db-per-service.md [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+
+Date: 2018-01-10
+
+## Status
+
+Proposed
+
+## Context
+
+Many of the components the digital business platform offers need a database for storing information.
```
Comment (cristina-sirbu) [inline docs/adrs/0008-using-one-db-per-service.md]: "Digital Business Platform"

File: docs/adrs/0008-using-one-db-per-service.md [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+Currently Content Services, Process Services and Sync Service do this but there might be more to come in the future. All three are using the same version of the database so the obvious question was raised: Can we just use one database for all our services?
+
+We identified advantages of using one database per service or one for all the services
```
Comment (cristina-sirbu) [inline docs/adrs/0008-using-one-db-per-service.md]: I think you've missed a punctuation mark here.

File: docs/adrs/0008-using-one-db-per-service.md [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+Choosing one database per service would put us in line with the current best practices in the microservices world, as well as allowing us to change versions of the db individually, and as an added bennefit it kills off the posibility of one service being able to impact performance on other services.
```
Comment (cristina-sirbu) [inline docs/adrs/0008-using-one-db-per-service.md]: I think it is "benefit", not "bennefit".

**Comment by montgolfiere** [comment]
I assume this decision means that each "service" can choose to use (ie. test, document & configure) it's own independent DB schema/namespace. However, I assume a DBP sys admin (deployer) may choose to configure multiple DB schemas/namespaces in the same DB instance, for compatible DB types/versions ?

In addition, do we have a recommended subset of DB *types* (including RDBMS vs NoSQL) that a service should choose from (unless agreed as an exception). This should minimise the chance that N services will mandate deployment to up to  N different DB types ?

As a historical example, the Desktop Sync "service" went through a number of project iterations (albeit, based on customer demand & requirements). Initially, it had to be re-implemented from MongoDB to PostgreSQL. Later still, additional work had to be done to allow compatibility with Oracle & also MySQL / MariaDB.

As of now, I assume the preferred options are:

RDBMS
- at least one of:
  - MySQL / MariaDB / AWS Aurora (MySQL-compatibility)
  - PostgreSQL / AWS Aurora (PostgreSQL-compatibility)
- also for Enterprise-only:
  - Oracle
  - Microsoft SQL Server
- optionally for dev-only:
  - H2

NoSQL
- Cassandra
- AWS DynamoDB

**Comment by svidrascu** [comment]
@montgolfiere 
Defining a set of db types is out of the scope of this ADR.
This only refers to using one database instance for one service within the default chart setup. Since the components can be configured to connect to external ones or different schema under an external ones there is no problem for a sys admin to customize the setup.

**Comment by montgolfiere** [comment]
@svidrascu @gavincornwell  Yes, thanks. Just wanted to make sure we capture that fact that if we have DB per service, then services may choose different DB types (polygot persistence). Hence, we should probably capture a new story to define a set of minimum / typical DB types used across (and integrated with) the DBP.

**Pull request merged**